### PR TITLE
[PUBDEV-3558] [PUBDEV-3413] [PUBDEV-3370] [PUBDEV-3115] Relocating h2o.jar distributed with the h2o python package

### PR DESCRIPTION
**Pull request by st-pasha**
Currently we put h2o.jar into the h2o_jar folder at the root of python libraries folder (i.e. outside the `h2o` package folder). This makes installation very fragile: sometimes h2o.jar is hard to find, or the wrong h2o.jar is used (especially on Windows machines), or it doesn't work at all with certain virtual environments, etc.

This change tries to mitigate the issue by putting h2o.jar into the h2o/backend/bin folder within the h2o package.

---
This change is [Reviewable](https://reviewable.io/reviews/h2oai/h2o-3/389)

### CODE REVIEW

**Comment by h2o-ops** [comment]
Tests passed!

**Comment by abal5** [comment]
Looks like good change to me.

File: h2o-py/conda/h2o/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,27 @@
+package:
+  name: h2o
+  version: "3.10.0.8"
```
Comment (mmalohlava) [inline h2o-py/conda/h2o/meta.yaml]: This need to be generated.

File: h2o-py/conda/h2o/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+
+source:
+  path: ../..
+
+requirements:
+  build:
+    - python
+    - setuptools
+    - colorama
```
Comment (mmalohlava) [inline h2o-py/conda/h2o/meta.yaml]: No version restrictions?

**Pull request merged**